.github/pull_request_template.md

```diff
@@ -0,0 +1,19 @@
+## Description
+<!--- Describe your changes in detail -->
+<!--- Why is this change required? What problem does it solve? -->
+
+## Testing information
+<!--- Please describe in detail how you tested your changes. -->
+<!--- This section should be through enough that reviewers can replicate the testing. -->
+
+## Issue link
+<!-- Add a link to the issue -->
+
+## Checklist before requesting a review
```

Comment:
These aren't applicable to every type of repository.

What are the added tests to a Terraform PR
What's the necessary observability to a GO package
What's the documentation for a Topo change


If this is a mental checklist before submitting a PR... can we write up our PR expectations.. document them in developer standards... and copy them into this template as a comment so the developer sees them when drafting the PR, but does not muddy up the PR with non-applicable text

Reply:
I don't think any of these can be applicable 100% of the time, no matter what the repository. If you're updating the readme, would you add tests?

It's pretty common to see these sorts of PR templates across github though, and if it's not applicable in whatever specific case, I think it's fair to just say that in the PR comment.

We already do have documentation for these standards, but people have not really adopted them. The documentation should be moved to the org readme though for sure. At this time, it is in confluence - https://revolutionparts.atlassian.net/wiki/spaces/~798050687/pages/2540405129/RevolutionParts+Engineering+Principles (yay sprawl). We did share this out though when it was written, and we've also talked about testing, observability, etc many times.

I think it's useful as a checklist for the person writing code and the reviewer as well.. These should be looked at as part of the review.

Reply:
PR to move engineering principles to github: https://github.com/encodium/.github-private/pull/4